cargo-dist/src/config.rs

```diff
@@ -65,6 +65,13 @@ pub struct DistMetadata {
     #[serde(default, with = "opt_string_or_vec")]
     pub ci: Option<Vec<CiStyle>>,
 
+    /// Vendor external actions.
+    ///
+    /// This will substantially increase the size of your
+    /// repository, but may be desirable in some security contexts.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub github_vendor_actions: Option<bool>,
```

Comment:
I wasn't sure how we wanted to handle naming this since users might want to handle non-github systems differently. I'm open to naming this in a generic way!